include/swift/AST/ConstTypeInfo.h

```diff
@@ -105,11 +105,19 @@ class BuilderValue : public CompileTimeValue {
 /// A dictionary literal value representation
 class DictionaryValue : public CompileTimeValue {
 public:
-  DictionaryValue() : CompileTimeValue(ValueKind::Dictionary) {}
+  DictionaryValue(std::vector<std::shared_ptr<CompileTimeValue>> elements)
+      : CompileTimeValue(ValueKind::Dictionary), elements(elements) {}
 
   static bool classof(const CompileTimeValue *T) {
     return T->getKind() == ValueKind::Dictionary;
   }
+
+  std::vector<std::shared_ptr<CompileTimeValue>> getElements() const {
+    return elements;
```

Comment:
Great points. Since we know `Dictionary` elements get extracted as `TupleValue`, we can definitely narrow the type earlier. Also, I'm fine with switching to that capitalization.